src/Microsoft.OpenApi.OData.Reader/PathItem/PathItemHandler.cs

```diff
@@ -101,5 +113,54 @@ protected virtual void AddOperation(OpenApiPathItem item, OperationType operatio
             IOperationHandler operationHander = provider.GetHandler(Path.Kind, operationType);
             item.AddOperation(operationType, operationHander.CreateOperation(Context, Path));
         }
+
+        /// <summary>
+        /// Set the parameters information for <see cref="OpenApiPathItem"/>
+        /// </summary>
+        /// <param name="item">The <see cref="OpenApiPathItem"/>.</param>
+        protected virtual void SetParameters(OpenApiPathItem item)
+        {
+            foreach (ODataKeySegment keySegment in Path.OfType<ODataKeySegment>())
+            {
+                IDictionary<string, string> mapping = ParameterMappings[keySegment];
+                foreach (var parameter in Context.CreateKeyParameters(keySegment, mapping))
+                {
+                    AppendParameter(item, parameter);
+                }
+            }
+
+            // Add the route prefix parameter v1{data}
+            if (Context.Settings.RoutePathPrefixProvider != null && Context.Settings.RoutePathPrefixProvider.Parameters != null)
+            {
+                foreach (var parameter in Context.Settings.RoutePathPrefixProvider.Parameters)
+                {
+                    item.Parameters.Add(parameter);
+                }
+            }
+        }
+
+        protected static void AppendParameter(OpenApiPathItem item, OpenApiParameter parameter)
```

Comment:
I believe we can refactor out this method (and the `SetParameters(OpenApiPathItem item)` method) into a common module that can be re-used by the `OperationHandler.cs` class as well.

Reply:
Yes. I am working on it.

Reply:
@irvinesunday the shared code has been refactored into a common module and re-used for both `OperationHandler.cs` and `PathItemHandler.cs`